relay-spans/src/span.rs

```diff
@@ -0,0 +1,459 @@
+    /// dropped_attributes_count is the number of dropped attributes. If the value is 0,
+    /// then no attributes were dropped.
+    #[serde(default)]
+    pub dropped_attributes_count: u32,
+    /// name of the event.
+    /// This field is semantically required to be set to non-empty string.
+    pub name: String,
+    /// time_unix_nano is the time the event occurred.
+    #[serde(default)]
+    pub time_unix_nano: u64,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Link {
+    /// attributes is a collection of attribute key/value pairs on the link.
+    /// Attribute keys MUST be unique (it is not allowed to have more than one
+    /// attribute with the same key).
+    #[serde(default)]
+    pub attributes: Vec<KeyValue>,
+    /// dropped_attributes_count is the number of dropped attributes. If the value is 0,
+    /// then no attributes were dropped.
+    #[serde(default)]
+    pub dropped_attributes_count: u32,
+    /// A unique identifier for the linked span. The ID is an 8-byte array.
+    #[serde(default)]
+    pub span_id: String,
+    /// A unique identifier of a trace that this linked span is part of. The ID is a
+    /// 16-byte array.
+    #[serde(default)]
+    pub trace_id: String,
+    /// The trace_state associated with the link.
+    #[serde(default)]
+    pub trace_state: String,
+}
+
+#[derive(Clone, Default, Debug, Deserialize_repr)]
+#[repr(u32)]
+pub enum SpanKind {
+    /// Unspecified. Do NOT use as default.
+    /// Implementations MAY assume SpanKind to be INTERNAL when receiving UNSPECIFIED.
+    Unspecified = 0,
+    /// Indicates that the span represents an internal operation within an application,
+    /// as opposed to an operation happening at the boundaries. Default value.
+    #[default]
+    Internal = 1,
+    /// Indicates that the span covers server-side handling of an RPC or other
+    /// remote network request.
+    Server = 2,
+    /// Indicates that the span describes a request to some remote service.
+    Client = 3,
+    /// Indicates that the span describes a producer sending a message to a broker.
+    /// Unlike CLIENT and SERVER, there is often no direct critical path latency relationship
+    /// between producer and consumer spans. A PRODUCER span ends when the message was accepted
+    /// by the broker while the logical processing of the message might span a much longer time.
+    Producer = 4,
+    /// Indicates that the span describes consumer receiving a message from a broker.
+    /// Like the PRODUCER kind, there is often no direct critical path latency relationship
+    /// between producer and consumer spans.
+    Consumer = 5,
+}
+
+#[derive(Clone, Debug, Deserialize, Default)]
+pub struct Status {
+    /// A developer-facing human readable error message.
+    #[serde(default)]
+    pub message: String,
+    /// The status code.
+    #[serde(default)]
+    pub code: StatusCode,
+}
+
+#[derive(Clone, Debug, Deserialize_repr, Default, PartialEq)]
+#[repr(u32)]
```

Comment:
```suggestion
#[repr(u8)]
```

Any reason for `u32` here?

Reply:
I was about to say I wanted to stick with what was [generated](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-proto/src/proto/grpcio/opentelemetry.proto.trace.v1.rs#L305C1-L305C1) but turns out, I'm even changing that to unsigned integer, so might as well use a `u8`.